### PR TITLE
Introduce extra sampling rate of 20% for configuration telemetry events

### PR DESCRIPTION
**Pull request by plousada**
### What and why?
This change applies extra sampling rate (20%) to the configuration telemetry events on top of the telemetry sampling rate configured.

### How?

A brief description of implementation details of this PR.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: Sources/Datadog/RUM/RUMTelemetry.swift [hunk 1]
```diff
@@ -48,7 +49,8 @@ internal final class RUMTelemetry: Telemetry {
         dateProvider: DateProvider,
         configurationEventMapper: RUMTelemetryConfiguratoinMapper?,
         delayedDispatcher: RUMTelemetryDelayedDispatcher?,
-        sampler: Sampler
+        sampler: Sampler,
+        configurationExtraSampler: Sampler = Sampler(samplingRate: 0.2) // Extra sample of 20% for configuration events
```
Comment (plousada) [inline Sources/Datadog/RUM/RUMTelemetry.swift:53]: I'm not sure if this is the most Swift idiomatic way.
Happy to take feedback.

Comment (maxep) [reply Sources/Datadog/RUM/RUMTelemetry.swift:53]: It is 👍

File: Tests/DatadogTests/Datadog/RUM/RUMTelemetryTests.swift [hunk 2]
```diff
@@ -266,6 +266,22 @@ class RUMTelemetryTests: XCTestCase {
         XCTAssertEqual(events.count, 0)
     }
 
+    func testSampledTelemetry_rejectAllConfiguration() throws {
+        // Given
+        let telemetry: RUMTelemetry = .mockWith(core: core, sampler: .mockKeepAll(), configurationExtraSampler: .mockRejectAll())
+
+        // When
+        // sends 10 telemetry events
+        for index in 0..<10 {
```
Comment (maxep) [inline Tests/DatadogTests/Datadog/RUM/RUMTelemetryTests.swift:275]: CI is complaining that this index isn't used, you can do:
```suggestion
        for _ in 0..<10 {
```

**Review by maxep — APPROVED** [review]
Looks great, thanks!

**Pull request merged**